src/boot/efi/boot.c

```diff
@@ -988,18 +988,27 @@ static BOOLEAN menu_run(
         return run;
 }
 
-static void config_add_entry(Config *config, ConfigEntry *entry) {
+static EFI_STATUS config_add_entry(Config *config, ConfigEntry *entry) {
         assert(config);
         assert(entry);
 
         if ((config->entry_count & 15) == 0) {
+                ConfigEntry **new_entries;
                 UINTN i = config->entry_count + 16;
-                config->entries = ReallocatePool(
+
+                new_entries = ReallocatePool(
```

Comment:
hmm, have a link?

Reply:
https://github.com/vathpela/gnu-efi/blob/269ef9dbc77ebec2723e0e6ae082bbca9516f5f1/lib/misc.c#L57

Reply:
christ... I guess they could confused, because they suppress the AllocatePool if newsize == 0...

But I guess thankfully, once we do abortive allocation we don#t have to care anymore...

Reply:
Well, no need for new_entries if config->entries gets clobbered anyway.